### PR TITLE
soft delete cascade policy

### PR DESCRIPTION
**Pull request by gibaneido**
Hi, this PR implements soft delete cascade over 0.4.x.
Tks

### CODE REVIEW

File: safedelete/utils.py [hunk 1]
```diff
@@ -16,3 +19,19 @@ def can_hard_delete(obj):
     collector = Collector(using=router.db_for_write(obj))
     collector.collect([obj])
     return not bool(collector.fast_deletes)
+
+
+def related_objects(obj):
+    """ Return a generator to the objects that would be deleted if we delete "obj" (excluding obj) """
+
+    collector = NestedObjects(using=router.db_for_write(obj))
```
Comment (Gagaro) [inline safedelete/utils.py]: I'm not sure we should get all the nested related objects there. We should only get the directly related ones, which should also cause the deletion of their related ones if they have `SOFT_DELETE_CASCADE` as well.

File: CHANGES [hunk 2]
```diff
@@ -2,6 +2,14 @@
 CHANGELOG
 =========
 
+0.4.1 (unreleased)
```
Comment (Gagaro) [inline CHANGES]: The 0.4.0 is not released yet, move this under the 0.4.0 section, thanks.

**Comment by Gagaro** [comment]
Also, can you add the ne wpolicy in the documentation? I forgot that for the 0.3.0 branch, we should at least do it here.

Thanks.

**Comment by pkainz** [comment]
@Gagaro 
Is there any update on this feature implementation?

**Comment by Gagaro** [comment]
Hi, I'll add it after #56. I'll try to release the 0.4.x then.

**Pull request closed without merging**